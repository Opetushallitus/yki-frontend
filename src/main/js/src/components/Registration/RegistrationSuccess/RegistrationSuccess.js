import React from 'react';
import PropTypes from 'prop-types';
import { withNamespaces } from 'react-i18next';

import classes from './RegistrationSuccess.module.css';
import Hyperlink from '../../UI/Hyperlink/Hyperlink';

export const registrationSuccess = props => {
  return (
    <div className={classes.RegistrationSuccess}>
      <div>
        <h1 data-cy="registration-success-header">
          {props.t('registration.success.header')}
        </h1>
        <p>{props.t('registration.success.info1')}:</p>
      </div>
      <div>
        <p>
          {props.t('registration.success.info2')} {props.formData.email}
        </p>
      </div>
      <div className={classes.InfoBox}>
        <p>
          <b>{props.t('registration.success.info3')}</b>{' '}
          {props.t('registration.success.info4')}
        </p>
      </div>
      <div className={classes.BackButton}>
        <Hyperlink to={'/yki/'} text={props.t('errorBoundary.return')} />
      </div>
    </div>
  );
};

registrationSuccess.propTypes = {
  initData: PropTypes.object.isRequired,
  formData: PropTypes.object.isRequired,
};

export default withNamespaces()(registrationSuccess);
