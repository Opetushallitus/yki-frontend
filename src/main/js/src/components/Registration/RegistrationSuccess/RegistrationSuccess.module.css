.RegistrationSuccess {
  width: auto;
  max-width: 100vw;
  margin: 0 14vw 10vh 14vw;
  min-height: 10vh;
}

.RegistrationSuccess p {
  font: normal normal normal 16px/24px PT Sans;
  color: var(--primary-blue);
}

.InfoBox p {
  margin-left: 8px;
  margin-right: 8px;
}
