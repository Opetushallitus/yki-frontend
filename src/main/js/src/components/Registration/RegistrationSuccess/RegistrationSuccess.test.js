import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import { registrationSuccess as RegistrationSuccess } from './RegistrationSuccess';

configure({ adapter: new Adapter() });

const examSession = 
  {
    registration_end_date: '2025-12-15',
    session_date: '2028-05-30',
    participants: 0,
    max_participants: 30,
    registration_start_date: '2028-03-01',
    language_code: 'fin',
    level_code: 'PERUS',
  };

describe('<RegistrationSuccess />', () => {
  it('should render registration success page', () => {
    const wrapper = shallow(
      <RegistrationSuccess t={key => key} initData={examSession} formData={{user: {email: 'test@test.com'}}}/>,
    );
   expect(toJson(wrapper)).toMatchSnapshot();
  });
});
