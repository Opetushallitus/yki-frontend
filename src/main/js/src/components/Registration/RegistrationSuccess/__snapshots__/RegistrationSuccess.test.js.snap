// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegistrationSuccess /> should render registration success page 1`] = `
<main
  id="main"
>
  <HeadlineContainer
    headlineContent={
      <ExamDetailsCard
        exam={
          Object {
            "exam_fee": "100.00",
            "language_code": "fin",
            "level_code": "PERUS",
            "max_participants": 30,
            "participants": 0,
            "registration_end_date": "2025-12-15",
            "registration_start_date": "2028-03-01",
            "session_date": "2028-05-30",
          }
        }
        isFull={false}
        successHeader={true}
      />
    }
    headlineImage="ophYki_image2.png"
    headlineTitle="registration.success.header"
  />
  <div
    className="RegistrationSuccess"
  >
    <backButton
      buttonText="errorBoundary.return"
      href="/yki/"
    />
    <div
      data-cy="registration-success"
    >
      <p>
        registration.success.info1
        :
         
        <b>
          undefined, undefined
          .
        </b>
      </p>
    </div>
    <div>
      <p>
        registration.success.info2
         
      </p>
    </div>
    <div
      data-cy="success-details-extra"
    >
      <p>
        <b>
          registration.success.info3
        </b>
         
        registration.success.info4
      </p>
    </div>
  </div>
</main>
`;
